pppd/sys-linux.c

```diff
@@ -300,108 +300,143 @@ extern int dfl_route_metric;
 
 
 /*
- * send_rtnetlink_msg - send rtnetlink message and return received error code:
+ * rtnetlink_msg - send rtnetlink message, receive response
+ * and return received error code:
  * 0              - success
  * positive value - error during sending / receiving message
  * negative value - rtnetlink responce error code
  */
-static int send_rtnetlink_msg(const char *desc, void *nlreq, size_t nlreq_len)
+static int rtnetlink_msg(const char *desc, int *shared_fd, void *nlreq, size_t nlreq_len, void *nlresp_data, size_t *nlresp_size, unsigned nlresp_type)
 {
-    struct {
+    struct nlresp_hdr {
         struct nlmsghdr nlh;
         struct nlmsgerr nlerr;
-    } nlresp;
+    } nlresp_hdr;
     struct sockaddr_nl nladdr;
-    struct iovec iov;
+    struct iovec iov[2];
```

Comment:
This variable is now used inconsistently in the code, sometimes iov is used, other times iov[i] ...

Reply:
I think after all changes, it should be consistent. `iov` is always array and if passing just one member of array I used `iov[i]` and when passing pointer (to array) then I used `iov`.